Shared/Samples/Generate offline map with custom parameters/GenerateOfflineMapWithCustomParametersView.swift

```diff
@@ -0,0 +1,142 @@
+                           let progress = model.generateOfflineMapJob?.progress {
+                            VStack(spacing: 16) {
+                                ProgressView(progress)
+                                    .progressViewStyle(.linear)
+                                    .frame(maxWidth: 200)
+                                
+                                Button("Cancel") {
+                                    isCancellingJob = true
+                                }
+                                .disabled(isCancellingJob)
+                                .task(id: isCancellingJob) {
+                                    guard isCancellingJob else { return }
+                                    // Cancels the job.
+                                    await model.cancelJob()
+                                    // Sets cancelling the job and generating an
+                                    // offline map to false.
+                                    isCancellingJob = false
+                                    isGeneratingOfflineMap = false
+                                }
+                            }
+                            .padding()
+                            .background(.regularMaterial)
+                            .clipShape(RoundedRectangle(cornerRadius: 15))
+                            .shadow(radius: 3)
+                        }
+                    }
+                    .overlay(alignment: .top) {
+                        Text("Offline map generated.")
+                            .frame(maxWidth: .infinity, alignment: .center)
+                            .padding(8)
+                            .background(.thinMaterial, ignoresSafeAreaEdges: .horizontal)
+                            .opacity(model.offlineMap != nil ? 1 : 0)
+                    }
+                    .toolbar {
+                        ToolbarItem(placement: .bottomBar) {
+                            Button("Generate Offline Map") {
+                                isShowingSetCustomParameters.toggle()
+                            }
+                            .disabled(model.isGenerateDisabled || isGeneratingOfflineMap)
+                            .popover(isPresented: $isShowingSetCustomParameters) {
+                                // Creates a rectangle from the area of interest.
+                                let viewRect = geometry.frame(in: .local).inset(
+                                    by: UIEdgeInsets(
+                                        top: 20,
+                                        left: geometry.safeAreaInsets.leading + 20,
+                                        bottom: 44,
+                                        right: -geometry.safeAreaInsets.trailing + 20
+                                    )
+                                )
+                                // Creates an envelope from the rectangle.
+                                if let extent = mapView.envelope(fromViewRect: viewRect) {
```

Comment:
Consider moving this logic into the "Generate Offline Map" button above. That way, the view hierarchy can be preserved here.